tensorzero_internal/src/model.rs

```diff
@@ -803,6 +804,66 @@ pub enum Credential {
     Missing,
 }
 
+/// Gets the value from a `OnceLock` or initializes it with the result of `f`
+/// If this is called simultaneously from multiple threads, it may call `f` multiple times
+/// If `f` returns an error, the `OnceLock` will remain uninitialized
+fn racy_get_or_try_init<T: Clone, E>(
```

Comment:
absolute nit here but I think this would be more readable if it was in the order these get called rather than bottom to top. Can you please rearrange and maybe write a general comment on the structure of this part of the code? So that future devs can easily understand.